proxy_list.go

```diff
@@ -0,0 +1,332 @@
+	request := codec.EncodeListRemoveWithIndexRequest(l.name, index)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return nil, err
+	}
+	return l.convertToObject(codec.DecodeListRemoveWithIndexResponse(response))
+}
+
+// RemoveAll removes the given elements from the list.
+// Returns true if the list has changed as the result of this operation, false otherwise.
+func (l *List) RemoveAll(elements ...interface{}) (bool, error) {
+	elementsData, err := l.validateAndSerializeValues(elements...)
+	if err != nil {
+		return false, err
+	}
+	request := codec.EncodeListCompareAndRemoveAllRequest(l.name, elementsData)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return false, err
+	}
+	return codec.DecodeListCompareAndRemoveAllResponse(response), nil
+}
+
+// RetainAll removes all elements from this list except the ones contained in the given slice.
+// Returns true if the list has changed as a result of this operation, false otherwise.
+func (l *List) RetainAll(elements ...interface{}) (bool, error) {
+	elementsData, err := l.validateAndSerializeValues(elements...)
+	if err != nil {
+		return false, err
+	}
+	request := codec.EncodeListCompareAndRetainAllRequest(l.name, elementsData)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return false, err
+	}
+	return codec.DecodeListCompareAndRetainAllResponse(response), nil
+}
+
+// Set replaces the element at the specified index in this list with the specified element.
+// Returns the previous element from the list.
+func (l *List) Set(index int32, element interface{}) (interface{}, error) {
+	elementData, err := l.validateAndSerialize(element)
+	if err != nil {
+		return nil, err
+	}
+	request := codec.EncodeListSetRequest(l.name, index, elementData)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return nil, err
+	}
+	return l.convertToObject(codec.DecodeListSetResponse(response))
+}
+
+// Size returns the number of elements in this list.
+func (l *List) Size() (int32, error) {
+	request := codec.EncodeListSizeRequest(l.name)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return 0, err
+	}
+	return codec.DecodeListSizeResponse(response), nil
+}
+
+// SubList returns a view of this list that contains elements between index numbers
+// from start (inclusive) to end (exclusive).
+func (l *List) SubList(start int32, end int32) ([]interface{}, error) {
+	request := codec.EncodeListSubRequest(l.name, start, end)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return nil, err
+	}
+	return l.convertToObjects(codec.DecodeListSubResponse(response))
+}
+
+// ToSlice returns a slice that contains all elements of this list in proper sequence.
+func (l *List) ToSlice() ([]interface{}, error) {
```

Comment:
Could you rename this function to `GetAll` to keep the name-compatibility within the client and with other clients?

Reply:
Renamed into `Iterator()` for the sake of consistency with `Queue.Iterator()`.